### PR TITLE
Limit the number of task headers per requestor in taskkeeper

### PR DESCRIPTION
**Pull request by Wolf480pl**
This implements (part of) #665

### CODE REVIEW

File: tests/golem/task/test_taskkeeper.py [hunk 1]
```diff
@@ -231,10 +231,58 @@ def test_is_correct(self):
         assert not correct
         assert err == "Subtask timeout is less than 0"
 
+    def test_task_limit(self):
```
Comment (marmistrz) [inline tests/golem/task/test_taskkeeper.py]: If you add `-> None`, you get mypy type-checks for free.

File: tests/golem/task/test_taskkeeper.py [hunk 2]
```diff
@@ -251,9 +299,11 @@ def get_dict_task_header():
 
 def get_task_header():
```
Comment (marmistrz) [inline tests/golem/task/test_taskkeeper.py]: How about adding `-> TaskHeader` to get free typechecks?

File: tests/golem/task/test_taskkeeper.py (hunk 1 above)
Comment (Wolf480pl) [reply tests/golem/task/test_taskkeeper.py]: Even in tests?

File: tests/golem/task/test_taskkeeper.py (hunk 2 above)
Comment (Wolf480pl) [reply tests/golem/task/test_taskkeeper.py]: Even in tests?

Comment (marmistrz) [reply tests/golem/task/test_taskkeeper.py]: Oh, I didn't notice these are tests. It's up to you then, but I prefer to have free checks in tests too. A common reason for failing tests is buggy tests :)

Comment (Wolf480pl) [reply tests/golem/task/test_taskkeeper.py]: At least tests don't fail silently.

**Comment by gborowik** [comment]
@Wolf480pl Could you please refactor the code by providing functions with as little code as possible as develop a separate test for the each of them? Such a process makes it human readable and usable in future. Thanks.

**Comment by Wolf480pl** [comment]
@gborowik is it better now?

**Comment by gborowik** [comment]
For sure. Please update the tests too.

**Comment by Wolf480pl** [comment]
@gborowik I added some tests

**Review by gborowik — APPROVED** [review]

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**